app/assets/javascripts/cocoon.js

```diff
@@ -118,15 +118,16 @@
     }, timeout);
   });
 
-
-  $(document).on("ready page:load turbolinks:load", function() {
+  var hideRemoveFields = function() {
     $('.remove_fields.existing.destroyed').each(function(i, obj) {
       var $this = $(this),
           wrapper_class = $this.data('wrapper-class') || 'nested-fields';
 
       $this.closest('.' + wrapper_class).hide();
     });
-  });
+  }
+  $(hideRemoveFields); // On page load
+  $(document).on('page:load turbolinks:load', hideRemoveFields); // On custom load events
 
```

Comment:
Hi @nathanvda 

I'm not sure if I understood why it should not work, but I don't have much experience with Turbolinks. To my understanding:

```
$(hideRemoveFields); // On page load
```
^ This will get executed on page load once regardless if turbolinks is used or not.

```
$(document).on('page:load turbolinks:load', hideRemoveFields); // On custom load events
```
^ This will get executed only when Turbolinks are used (either old version or the new one). It will get executed on initial load and subsequent turbolink-loads of body content.

Thus when Turbolinks is used the ```hideRemoveFields``` code will get executed twice on initial load. However it should not have any side effects.

Actually maybe this could be even better implementation:
```js
  $(function() {
    hideRemoveFields();
    $(document).on('page:load turbolinks:load', hideRemoveFields); // Turbolinks support
  });
```

This will work with all jQuery versions. It will also work with Turbolinks and the code will be executed once per page load (either initial or one initiated by Turbolinks).

Am I missing something?